meta.go

```diff
@@ -0,0 +1,100 @@
+package yarpc
+
+import (
+	"context"
+
+	"go.uber.org/yarpc/api/transport"
+)
+
+// CallOption defines options that may be passed in at call sites to other
+// services.
+//
+// These may be used to add or alter the request.
+type CallOption struct{ apply func(*Call) }
+
+type callHeader struct{ k, v string }
+
+// Call represents an outgoing call.
+//
+// It holds any per-call options for a request. Encoding authors may use Call
+// to hydrate Requests from call-site options.
+type Call struct {
+	// request attributes to fill if non-nil
+	headers         []callHeader
+	shardKey        *string
+	routingKey      *string
+	routingDelegate *string
+
+	// If non-nil, response headers should be written here.
+	responseHeaders *Headers
+}
+
+// WriteToRequest fills the given request with request-specific options from
+// the call.
+func (c *Call) WriteToRequest(ctx context.Context, req *transport.Request) {
```

Comment:
Intentionally. Right now, we're not reading anything from the context. I was
thinking about the scenario where some call-specific options are on the
context.

On that line of thought, is there a scenario where an option might change the
context itself? If so, maybe `WriteToRequest` should return the context. By
default it will just return it as-is but maybe some option in the future will
need to change it?